test/automate/automate-utils.js

```diff
@@ -14,285 +14,292 @@
-    return driver.wait(
+  const clickElement = locator => driver.wait(
       until.elementLocated(locator),
       elementTimeout
     )
-    .then(clickWhenVisible, errorFunc);
-  };
+    .then(clickWhenVisible);
 
   // Waits for an element to be visible, and then clicks it.
-  const clickWhenVisible = function(element) {
-    return driver.wait(
-     until.elementIsVisible(element),
-     elementTimeout
-   )
-   .then(function(element) {
-     element.click();
-     return element;
-   },
-   errorFunc);
-  };
+  const clickWhenVisible = element => driver.wait(
+      until.elementIsVisible(element),
+      elementTimeout
+    )
+    .then(element => element.click().then(() => element));
 
   // Waits for an element to have opacity of 1. Helps for waiting for elements
   // that fade in. Returns the element when the opacity reaches '1' so that
   // this can be chained with other promises.
-  const waitOpacity = function(element) {
-    return driver.wait(function() {
-      return element.getCssValue('opacity')
-      .then(function(opacity) {
-        if (opacity === '1') {
-          return element;
-        } else {
-          return false;
-        }
-      });
-    });
-  };
+  const waitOpacity = element => driver.wait(() => (
+    element.getCssValue('opacity')
+    .then(opacity => (opacity === '1') ? element : false)
+  ));
 
   // Handles errors. Saves a screenshot so we can see what the error is.
-  const errorFunc = function(error) {
-    // Ignore unexpected alert error, log all others.
-    if (error.name === 'UnexpectedAlertOpenError') {
-      console.log('Ignoring UnexpectedAlertOpenError');
-      saveScreenshot(driver, screenshotDir, name, `ERROR-${error.name}`);
-    } else {
-      saveScreenshot(driver, screenshotDir, name, `ERROR-${error.name}`);
+  const errorFunc = error => {
       console.error(error);
-    }
+      return saveScreenshot(name, `ERROR-${error.name}`);
+  };
+
+  // Alert appears only once.
+  let accepted = false;
+  const acceptAlert = () => {
+    if (accepted) { return null; }
+    return driver.wait(until.alertIsPresent(), 10000)
+      .then(
+        () => {
+          accepted = true;
+          return driver.switchTo().alert().accept();
+        },
+        () => 'do nothing'
+      );
   };
 
   // Test automation starts here.
 
-  // Mac OS for some reason doesn't like the 'start-maximized' flag.
-  // In some cases (e.g. Mac, headless Chrome) the "start-maximized" flag is
-  // ignored, so we do this just in case.
-  driver.manage().window().setPosition(0, 0);
-  driver.manage().window().setSize(browserMaxWidth, browserMaxHeight);
+  // Firefox doesn't have an option to maximize the window
+  // So we set the size here.
+  return driver.manage().window().setPosition(0, 0)
+  .then(() => driver.manage().window()
+    .setSize(browserMaxWidth, browserMaxHeight)
+  )
 
   // Get the specified URL.
-  driver.get(URL);
+  .then(() => driver.get(URL))
+
+  // Will be not needed after deploy
```

Comment:
What does this mean "will not be needed after deploy"?

Reply:
For now, in the headless mode, an alert is displayed for both Firefox and Chrome. After deployment, the alert will not be displayed. So this check will be not needed